runtime/http_client.go

```diff
@@ -75,17 +75,18 @@ func NewHTTPClientContext(
 	logger *zap.Logger,
 	ContextMetrics ContextMetrics,
 	clientID string,
-	methodNames []string,
+	methodToTargetEndpoint map[string]string,
 	baseURL string,
 	defaultHeaders map[string]string,
 	timeout time.Duration,
 ) *HTTPClient {
-	loggers := make(map[string]*zap.Logger, len(methodNames))
+	loggers := make(map[string]*zap.Logger, len(methodToTargetEndpoint))
 
-	for _, methodName := range methodNames {
+	for methodName, targetEndpointName := range methodToTargetEndpoint {
 		loggers[methodName] = logger.With(
 			zap.String("clientID", clientID),
 			zap.String("clientMethod", methodName),
+			zap.String("targetEndpoint", targetEndpointName),
```

Comment:
```suggestion
			zap.String(logFieldClientThriftMethod, targetEndpointName),
```

Reply:
This suggestion is to align with tchannel client log fields, https://github.com/uber/zanzibar/blob/master/runtime/tchannel_client.go#L115-L118